src/bp/ui-studio/src/web/views/FlowBuilder/diagram/Forms/ExecuteForm/index.tsx

```diff
@@ -0,0 +1,225 @@
+import { Button, Tooltip } from '@blueprintjs/core'
+import { ExecuteNode, FlowNode } from 'botpress/sdk'
+import {
+  Form,
+  Icons,
+  lang,
+  MainLayout,
+  MoreOptions,
+  MoreOptionsItems,
+  MultiLevelDropdown,
+  sharedStyle,
+  Tabs
+} from 'botpress/shared'
+import cx from 'classnames'
+import { CUSTOM_ACTION } from 'common/action'
+import { ControlType, CustomTemplate } from 'common/controls'
+import { LocalActionDefinition } from 'common/typings'
+import _ from 'lodash'
+import React, { FC, Fragment, useEffect, useState } from 'react'
+
+import style from './style.scss'
+import ConfigAction from './ConfigAction'
+
+interface Props {
+  node: FlowNode
+  customKey: string
+  contentLang: string
+  actions: LocalActionDefinition[]
+  events: any
+  formData: ExecuteNode
+  onUpdate: (data: Partial<ExecuteNode>) => void
+  deleteNode: () => void
+  close: () => void
+  onCodeEdit: (value: string | undefined, onChange: (data: string) => void, template?: string | CustomTemplate) => void
+  keepOpen: boolean
+}
+
+const newAction = { label: 'Code New Action', value: CUSTOM_ACTION }
+
+const executeReducer = (state, action) => {
+  if (action.type === 'setup') {
+    return {
+      ...state,
+      ...action.data
+    }
+  } else if (action.type === 'updateArgs') {
+    return {
+      ...state,
+      args: action.data
+    }
+  } else if (action.type === 'changeAction') {
+    return {
+      ...state,
+      selectedAction: action.data,
+      code: ''
+    }
+  }
+}
+
+const ExecuteForm: FC<Props> = ({
```

Comment:
correct me if I'm wrong but this won't work if code editor isn't enabled. correct ?

Reply:
Yup, because we use the exported component from the code editor